File: hola_cliente.js
var thrift = require('thrift');
var HolaServicio = require('./gen-nodejs/HolaServicio');
var ttypes = require('./gen-nodejs/hola_types');
const assert = require('assert');

var transport = thrift.TBufferedTransport;
var protocol = thrift.TBinaryProtocol;

var connection = thrift.createConnection("localhost", 9090, {
  transport : transport,
  protocol : protocol
});

connection.on('error', function(err) {
  assert(false, err);
});

var client = thrift.createClient(HolaServicio, connection);

client.hola_func(function(err, response) {
  console.log(response);
  connection.end();
});

File: hola_cliente.rb
$:.push('gen-rb')
$:.unshift '../../lib/rb/lib'

require 'thrift'

require 'hola_servicio'

begin
    port = ARGV[0] || 9090

    transport = Thrift::BufferedTransport.new(Thrift::Socket.new('localhost', port))
    protocol = Thrift::BinaryProtocol.new(transport)
    client = HolaServicio::Client.new(protocol)

    transport.open()

    mensaje = client.hola_func()
    puts"[Cliente] Se recibio " + mensaje

    transport.close()

rescue Thrift::Exception => tx
  print 'Thrift::Exception: ', tx.message, "\n"
end
